src/app-layout/visual-refresh/drawers.tsx

```diff
@@ -265,6 +359,16 @@ export function MobileTriggers() {
     return null;
   }
 
+  const overflowItemHasBadge = () => {
+    const overflowItems = drawers?.items.slice(1, drawers.items.length);
+    return overflowItems && overflowItems.filter(item => item.badge).length > 0;
+  };
+
+  const mobileItems = !toolsHide && tools ? drawers.items.slice(0, 1) : drawers.items.slice(0, 2);
+  const hasOverflowMenu = !toolsHide && tools ? drawers?.items?.length > 2 : drawers?.items?.length > 3;
+  const overflowItems =
+    !toolsHide && tools ? drawers.items.slice(1, drawers.items.length) : drawers.items.slice(2, drawers.items.length);
```

Comment:
I think this code could also reuse `splitItems` helper, if it would work like this

```
function splitItems(allItems: Array<DrawerItem>, splitIndex: number) {...}

// use in desktop mode
splitItems(drawers.items, getIndexOfOverflowItem());

// use in mobile mode
splitItems(drawers.items, 2);
```